bones/fileBone.py

```diff
@@ -25,31 +25,37 @@ def getReferencedBlobs(self, valuesCache, name):
 		else:
 			raise ValueError("Unknown value for bone %s (%s)" % (name, str(type(valuesCache[name]))))
 
-	def unserialize( self, valuesCache, name, expando ):
-		res = super( fileBone, self ).unserialize( valuesCache, name, expando )
+	def unserialize(self, valuesCache, name, expando):
+		res = super(fileBone, self).unserialize(valuesCache, name, expando)
+		currentValue = valuesCache[name]
 		if not request.current.get().isDevServer:
 			# Rewrite all "old" Serving-URLs to https if we are not on the development-server
-			if isinstance(valuesCache[name], dict) and valuesCache[name]["dest"].get("servingurl"):
-				if valuesCache[name]["dest"]["servingurl"].startswith("http://"):
-					valuesCache[name]["dest"]["servingurl"] = valuesCache[name]["dest"]["servingurl"].replace("http://","https://")
-			elif isinstance( valuesCache[name], list ):
-				for val in valuesCache[name]:
-					if isinstance(val, dict) and val["dest"].get("servingurl"):
-						if val["dest"]["servingurl"].startswith("http://"):
-							val["dest"]["servingurl"] = val["dest"]["servingurl"].replace("http://","https://")
-		if isinstance(valuesCache[name], dict):
-			if not "mimetype" in valuesCache[name]["dest"] or not valuesCache[name]["dest"]["mimetype"]:
-				if "meta_mime" in valuesCache[name]["dest"] and valuesCache[name]["dest"]["meta_mime"]:
-					valuesCache[name]["dest"]["mimetype"] = valuesCache[name]["dest"]["meta_mime"]
-				elif "metamime" in valuesCache[name]["dest"] and valuesCache[name]["dest"]["metamime"]:
-					valuesCache[name]["dest"]["mimetype"] = valuesCache[name]["dest"]["metamime"]
-		elif isinstance(valuesCache[name], list):
-			for val in valuesCache[name]:
-				if not "mimetype" in val["dest"] or not val["dest"]["mimetype"]:
-					if "meta_mime" in val["dest"] and val["dest"]["meta_mime"]:
-						val["dest"]["mimetype"] = val["dest"]["meta_mime"]
-					elif "metamime" in val["dest"] and val["dest"]["metamime"]:
-						val["dest"]["mimetype"] = val["dest"]["metamime"]
+			if isinstance(currentValue, dict) and "servingurl" in currentValue["dest"]:
```

Comment:
I think these are not equal. I'm not sure if it's guaranteed that currentValue["dest"]["servingurl"] is always the empty string instead of None - then currentValueDest["servingurl"].startswith("http://") will fail.